apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -87,13 +88,31 @@ pub mod oracle_process {
 
     /// Set up an Oracle and run the process where the Oracle
     /// processes and forwards Ethereum events to the ledger
-    pub async fn run_oracle(
-        url: &str,
+    pub fn run_oracle(
+        url: impl AsRef<str>,
```

Comment:
What's the motivation for switching to `impl AsRef<str>`?

Reply:
Before we could use the `url: &str` reference directly, now we need an owned `url` string of some sort we can move into `tokio::task::spawn_blocking(...)`. We could also accept `url: String` or call `let url = url.to_owned()` and keep using `url: &str`. In actuality we were still passing `&str` at the sole call site, I updated it to pass the `ethereum_url` `String`.

Reply:
I see. Ok :+1: